base/range.jl

```diff
@@ -585,12 +585,13 @@ function show(io::IO, r::LinRange{T}) where {T}
     print(io, "LinRange{")
     show(io, T)
     print(io, "}(")
-    show(io, first(r))
-    print(io, ", ")
-    show(io, last(r))
-    print(io, ", ")
-    show(io, length(r))
-    print(io, ')')
+    ioc = IOContext(io, :typeinfo=>T)
+    show(ioc, first(r))
+    print(ioc, ", ")
+    show(ioc, last(r))
+    print(ioc, ", ")
+    show(ioc, length(r))
```

Comment:
I can't think of a concrete case where this would make a difference, but `r.len` isn't of type `T`, so:
```suggestion
    show(io, length(r))
```